### PR TITLE
Assignment6 1 2

### PR DESCRIPTION
**Pull request by kirsimarianne**

### CODE REVIEW

File: w6/assignment6_1_2/character.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    void set(float wavelength);
+
+    private:
+    Tag tag{};
+    union
+    {
+        int id;
+        char first_letter;
+        float wavelength;
+    };
```
Comment (kmoilane) [inline w6/assignment6_1_2/character.h:40]: Maybe more of a personal preference, but these could be tabbed once more to distinguish them from the `private:` for better readability. Same with the public above.

**Review by kmoilane — COMMENTED** [review]
Code looks good overall. However I think that you should be abled to change the tag by just changing the value using set(). So in the setters you should change the tag and the value. Other thing I noticed are the asserts that are currently in the main, but if I've understood correctly, they should be inside the getter functions before the return statement. That's how I understood the assignment, but I might be wrong as well! 😄  Good job otherwise!

File: w6/assignment6_1_2/character.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        this->wavelength = wavelength;
+    }
+    else
+    {
+        set_tag(Tag::not_found);
+    }
+}
+
+const int& Character::get_id() const
```
Comment (ronchaine) [inline w6/assignment6_1_2/character.cpp:71]: Check inside these that the correct type is trying to be acquired.  (e.g. throw if the tag is not ID when trying to get_id)

**Comment by ronchaine** [comment]
> Code looks good overall. However I think that you should be abled to change the tag by just changing the value using set(). So in the setters you should change the tag and the value. Other thing I noticed are the asserts that are currently in the main, but if I've understood correctly, they should be inside the getter functions before the return statement. That's how I understood the assignment, but I might be wrong as well! smile Good job otherwise!

Yup, this needs to be done still.

**Pull request merged**